### ISSUE TITLE
Strange error with invalid NTuple syntax

### ISSUE THREAD

**Issue by sbromberger** [JuliaLang/julia#28058]
```
julia> NTuple(3)
ERROR: UndefVarError: E not defined
Stacktrace:
 [1] _totuple(::Type{Tuple{Vararg{T,N}} where T where N}, ::Int64) at ./tuple.jl:259
 [2] Tuple{Vararg{T,N}} where T where N(::Int64) at ./tuple.jl:243
 [3] top-level scope at none:0
```

```
julia> versioninfo()
Julia Version 0.7.0-beta.256
Commit 324b30d2bd (2018-07-11 15:01 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin17.6.0)
  CPU: Intel(R) Core(TM) i7-4771 CPU @ 3.50GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, haswell)
Environment:
  JULIA_EDITOR = /usr/local/bin/code
```

**Comment by JeffBezanson** [comment]
This is one of those cases where an unhelpful error just falls out of "the way things work". `NTuple` is not syntax, it's an identifier. Tuple types can be used as constructors, e.g. `Tuple([1,2])`. So `NTuple(3)` tries to construct an instance of `NTuple` from `3`. That errors because the element type to use is not sufficiently specified by the type `NTuple`.

We can add a better error for this, or determine an element type somehow, e.g. if the argument is `HasEltype`.

**Comment by inkydragon** [comment]
This issue has been fixed in 1.6.3.

```julia
julia> NTuple(3)
ERROR: ArgumentError: too few elements for tuple type Tuple{Vararg{T, N}} where {N, T}
Stacktrace:
 [1] _totuple_err(T::Any)
   @ Base .\tuple.jl:309
 [2] _totuple
   @ .\tuple.jl:315 [inlined]
 [3] _totuple
   @ .\tuple.jl:316 [inlined]
 [4] (Tuple{Vararg{T, N}} where {N, T})(itr::Int64)
   @ Base .\tuple.jl:303
 [5] top-level scope
   @ REPL[2]:1

julia> versioninfo()
Julia Version 1.6.3
Commit ae8452a9e0 (2021-09-23 17:34 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i5-9400F CPU @ 2.90GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)
```

**Comment by KristofferC** [comment]
https://github.com/JuliaLang/julia/pull/31160 probably?
